### PR TITLE
RELATED: RAIL-1118 Propagate requested measure format from visualizationObject to AFM

### PR DESCRIPTION
**Pull request by ocetnik**
`8.1.1-ocetnik-doc-measure-format-2018-09-19T13-29-51-263Z`

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/83/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/94/ : FAILURE in 3m 46s

File: src/DataLayer/converters/MeasureConverter.ts [hunk 1]
```diff
@@ -22,8 +23,9 @@ function convertMeasure(measure: IMeasure): AFM.IMeasure {
 
     const convertedDefinition = convertMeasureDefinition(definition);
 
+    const requestedMeasureFormat = get(measure.measure, 'format', undefined);
```
Comment (martinmosko) [inline src/DataLayer/converters/MeasureConverter.ts]: its necessary to specify `undefined`? `lodash.get` by default its returning `undefined`

File: src/DataLayer/converters/tests/MeasureConverter.spec.ts [hunk 2]
```diff
@@ -16,9 +16,9 @@ describe('convertMeasure', () => {
         });
     });
 
-    it('should ignore format defined on simple measures', () => {
+    it('should convert simple measure with format', () => {
```
[the remark this answers is not in the document]
Comment (ocetnik) [reply src/DataLayer/converters/tests/MeasureConverter.spec.ts]: yes, it's `should convert simple measures`

File: src/DataLayer/converters/MeasureConverter.ts (hunk 1 above)
Comment (ocetnik) [reply src/DataLayer/converters/MeasureConverter.ts]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/85/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/96/ : SUCCESS in 1m 05s

File: src/DataLayer/converters/MeasureConverter.ts [hunk 3]
```diff
@@ -147,11 +148,15 @@ function convertPreviousPeriodMeasureDefinition(definition: IPreviousPeriodMeasu
...
-    if (VisualizationObject.isMeasureDefinition(definition)) {
-        return convertSimpleMeasureDefinitionFormat(definition);
-    }
-    return null;
+function getFormat(measure: IMeasure): string | undefined {
+    const { measure: { definition } } = measure;
+    const measureFormat = get(measure.measure, 'format');
+
+    const predefinedFormat = VisualizationObject.isMeasureDefinition(definition)
+        ? convertSimpleMeasureDefinitionFormat(definition)
```
Comment (martinmosko) [inline src/DataLayer/converters/MeasureConverter.ts]: hmm probably one extra change could you please change the function name to something else?

for example `getPredefinedFormat` ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/86/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/97/ : SUCCESS in 1m 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/87/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/98/ : SUCCESS in 40s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/88/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/100/ : SUCCESS in 1m 19s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/31/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/101/ : SUCCESS in 1m 22s

**Pull request merged**